### DISCUSSION TITLE
Errore sul totale dei pagamenti

### DISCUSSION THREAD

**Discussion by massi82** [fattureincloud/api#378] [Q&A] [answered]
Salve a tutti,

ho il seguente dizionario per la creazione di una fattura in python:

    invoice_data = { "data":{
      "type": "invoice",
      "year": 2023,
      "numeration": "",
      "subject": "",
      "visible_subject": "Servizi Logistici",
      "rc_center": "",
      "amount_rivalsa": 0,
      "amount_rivalsa_taxable": 0,
      "amount_global_cassa_taxable": 0,
      "amount_cassa": 0,
      "amount_cassa_taxable": 0,
      "amount_cassa2": 0,
      "amount_cassa2_taxable": 0,
      "amount_withholding_tax": 0,
      "amount_withholding_tax_taxable": 0,
      "amount_other_withholding_tax": 0,
      "amount_enasarco_taxable": 0,
      "amount_other_withholding_tax_taxable": 0,
      "ei_cassa_type": None,
      "ei_cassa2_type": None,
      "ei_withholding_tax_causal": None,
      "ei_other_withholding_tax_type": None,
      "ei_other_withholding_tax_causal": None,
      "stamp_duty": 0,
      "use_gross_prices": False,
      "e_invoice": True,
      "delivery_note": False,
      "accompanying_invoice": False,
      "amount_net": 366.6,
      "amount_vat": 103.4,
      "amount_gross": 470,
      "amount_due_discount": 0,
      "h_margins": 15,
      "v_margins": 16,
      "show_payment_method": True,
      "show_payments": True,
      "show_totals": "all",
      "show_paypal_button": False,
      "show_notification_button": False,
      "is_marked": False,
    #   "created_at": "2023-11-24 17:37:54",
    #   "updated_at": "2023-11-27 12:21:36",
      "entity": {
        "name": "....",
        "vat_number": "...",
        "tax_code": "...",
        "address_street": "...",
        "address_postal_code": "....",
        "address_city": "....",
        "address_province": "....",
        "address_extra": "",
        "country": "Italia",
        "certified_email": "",
        "ei_code": "U3NK44I",
        "entity_type": "client",
        "type": "company",
        "default_payment_terms": 0,
        "default_payment_terms_type": "standard",
        "shipping_address": "",
        "default_vat": None,
        "default_payment_method": None,
        "bank_name": "",
        "bank_iban": "",
        "default_discount": 0,
        "discount_highlight": False,
        "has_intent_declaration": False,
        "intent_declaration_protocol_number": None,
        "intent_declaration_protocol_date": None
      },
      "date": "2023-11-29",
      "currency": {
        "id": "EUR",
        "exchange_rate": "1.00000",
        "symbol": "€"
      },
      "language": {
        "code": "it",
        "name": "Italiano"
      },
      "notes": "",
      "payment_method": {
        "id": None,
        "name": "BONIFICO BANCARIO",
        "details": [
          {
            "title": "IBAN",
            "description": "..."
          },
          {
            "title": "BENEFICIARIO",
            "description": "..."
          },
          {
            "title": "",
            "description": ""
          },
          {
            "title": "",
            "description": ""
          },
          {
            "title": "",
            "description": ""
          }
        ]
      },
      "use_split_payment": False,
      "merged_in": None,
      "original_document": None,
      "items_list": [
        {
          "product_id": None,
          "code": "01",
          "name": "Ordine  del  02/11/2023",
          "measure": "",
          "category": "",
          "apply_withholding_taxes": False,
          "discount": 0,
          "discount_highlight": False,
          "in_dn": False,
          "qty": 1,
          "net_price": 366.6,
          "vat": {
            "id": 0,
            "value": 22,
            "description": ""
          },
          "stock": None,
          "description": "1 item",
          "gross_price": 470,
          "not_taxable": False,
          "ei_raw": None
        }
      ],
      "payments_list": [
        {
          "amount": 470,
          "due_date": "2024-01-31",
          "paid_date": None,
          "payment_terms": {
            "days": 60,
            "type": "end_of_month"
          },
          "status": "not_paid",
          "payment_account": None,
          "ei_raw": None
        }
      ],
      "extra_data": {
        "show_sofort_button": False,
        "multifatture_sent": None,
        "accountant_causal_code": None,
        "ts_communication": False,
        "ts_tipo_spesa": None,
        "ts_flag_tipo_spesa": None,
        "ts_pagamento_tracciato": None,
        "ts_opposizione": None,
        "ts_status": None,
        "ts_file_id": None,
        "ts_sent_date": None,
        "ts_full_amount": False,
        "imported_by": None,
        "ts_single_sending": None
      },
      "rivalsa": 0,
      "rivalsa_taxable": 100,
      "cassa": 0,
      "cassa2": 0,
      "global_cassa_taxable": 100,
      "cassa_taxable": 100,
      "cassa2_taxable": 100,
      "withholding_tax": 0,
      "withholding_tax_taxable": 100,
      "other_withholding_tax": 0,
      "other_withholding_tax_taxable": 100,
      "seen_date": None,
      "next_due_date": "2024-01-31",
      "template": {
        "id": 10,
        "name": "New Standard S1"
      },
      "show_tspay_button": False,
            "url": "abcd.pdf",
      "ei_data": {
        ....
      },
      "ei_raw": None,
      "ei_ts_data": {
        "status": 0,
        "id": None,
        "info": None
      },
      "ei_status": "not_sent",
      "locked": False,
      "has_ts_pay_pending_payment": False
    }
    } 
    
    Quando provo ad eseguire il comando:

issued_documents_api_instance.create_issued_document(company_id, invoice_data)

ho il seguente errore:

HTTP response body: {"error":{"message":"Il totale dei pagamenti non corrisponde al totale da pagare.","validation_result":null}}

Non mi è chiaro dove sia il disallineamento sui totali, mi potete aiutare?

**Comment by massi82** [comment]
Errore riproducibile anche con questo script python:

```
with fattureincloud_python_sdk.ApiClient(configuration) as api_client:
    
    # Retrieve the first company id
    user_api_instance = user_api.UserApi(api_client)
    user_companies_response = user_api_instance.list_user_companies()
    company_id = user_companies_response.data.companies[0].id
    print(company_id)
            
    entity = Entity(
        id=1,
        name="name",
        vat_number="vat",
        tax_code="tax",
        address_street="address",
        address_postal_code="postal_code",
        address_city="city",
        address_province="PR",
        country="Italia"
    )

    invoice = IssuedDocument(
        type = IssuedDocumentType("invoice"),
        entity = entity,
        date = datetime.date(2023, 11, 29),
        number = 1,
        numeration = "/fatt",
        subject = "internal subject",
        visible_subject = "visible subject",
        currency = Currency(
            id="EUR"
        ),
        language = Language(
            code="it",
            name="italiano"
        ),
        items_list = [
            IssuedDocumentItemsListItem(
                product_id=None,
                code="01",
                name="Ordine  del  02/11/2023",
                net_price=366.6,
                category="spedizione",
                discount=0.0,
                qty=1.0,
                vat=VatType(
                    id=0,
                    value=22
                )
            )
        ],
        payments_list = [
            IssuedDocumentPaymentsListItem(
                amount=470.0,
                due_date=datetime.date(2024, 1, 31),
                paid_date=None,
                status=IssuedDocumentStatus("not_paid"),
                payment_account=PaymentAccount(
                    id=...
                )
            )
        ],
        payment_method = PaymentMethod(
            id=...,
            bank_iban="IBAN",
            bank_name="Bank Name",
            bank_beneficiary="Beneficiario"
        ),
        # attachment_token = "YmMyNWYxYzIwMTU3N2Y4ZGE3ZjZiMzg5OWY0ODNkZDQveXl5LmRvYw",
        # template = DocumentTemplate(
        #     id=150
        # )
    )
    create_issued_document_request = CreateIssuedDocumentRequest(
        data = invoice
    )
    api_instance = issued_documents_api.IssuedDocumentsApi(api_client)
    try:
        api_response = api_instance.create_issued_document(company_id, create_issued_document_request=create_issued_document_request)
        pprint(api_response)
    except fattureincloud_python_sdk.ApiException as e:
        print("Exception when calling IssuedDocumentsApi->create_issued_document: %s\n" % e)
```

**Comment by massi82** [comment, marked as answer]
Problema risolto, avevo tolto l'Iva dal totale anzichè aggiungerla al netto.

Chiudo la conversazione.

**Reply by valmoz** [reply]
Buongiorno @massi82 
perfetto. 

Nel caso in cui si dovesse ripresentare lo stesso problema, abbiamo preparato una guida a riguardo: [Invoice Totals](https://developers.fattureincloud.it/docs/guides/invoice-totals)

Grazie
Mauro

**Reply by valmoz** [reply]
Dimenticavo, la pregherei anche di modificare i suoi messaggi rimuovendo gli ID (se reali) e tutte le altre eventuali informazioni sensibili, essendo questa una pagina pubblica.

Grazie

**Reply by massi82** [reply]
Certamente, fatto. 

Grazie.
